backpack/core/derivatives/linear.py

```diff
@@ -14,43 +18,204 @@ class LinearDerivatives(BaseParameterDerivatives):
     * i: Input dimension
     """
 
-    def hessian_is_zero(self):
+    def hessian_is_zero(self) -> bool:
+        """Linear layer output is linear w.r.t. to its input.
+
+        Returns:
+            True
+        """
         return True
 
-    def _jac_t_mat_prod(self, module, g_inp, g_out, mat):
-        """Apply transposed Jacobian of the output w.r.t. the input."""
-        d_input = module.weight.data
-        return einsum("oi,vno->vni", (d_input, mat))
+    def _jac_t_mat_prod(
+        self, module: Linear, g_inp: Any, g_out: Any, mat: Tensor
```

Comment:
I specified `Any` intentionally, since `g_inp` and `g_out` are not required by the implementation (see docstring), and this function will work no matter what is passed to those arguments. I have no specific opinion how to proceed in such cases. Do you happen to know guidelines how to type-annotate? Otherwise I can go ahead and restrict the types as you requested.